functions/access.js

```diff
@@ -114,16 +116,16 @@ const createToken = async (venueId, uid, password, email, code) => {
       ]);
 
       if (!venue.exists) {
-        return undefined;
+        throw new HttpsError("not-found", `venue ${venueId} does not exist`);
       }
 
       const token = uuid();
 
       const tokenData = {
         usedAt: [Date.now()],
-        password: password,
-        email: email,
-        code: code,
+        ...(password && { ...password }),
+        ...(email && { ...email }),
+        ...(code && { ...code }),
```

Comment:
Are `password`/`email`/`code` intended to be objects here? (implied by the 'internal spread', eg `{ ...code }`). I would have expected them to be strings or similar.

Reply:
They are strings, stupid mistake. Good catch.